client/modules/IDE/components/NewFileModal.jsx

```diff
@@ -78,9 +80,16 @@ function validate(formProps) {
   return errors;
 }
 
+function mapDispatchToProps(dispatch) {
+  return {
+    setFieldValue: (fieldName, fieldValue) => {
+      dispatch(change('new-file', fieldName, fieldValue));
```

Comment:
Is this the correct approach?

Reply:
no, you should use `bindActionCreators` from `redux`